spec/models/loader/types.rb

```diff
@@ -0,0 +1,22 @@
+require 'spec_helper'
+
+describe Loader::Types do
+
+  let(:new_user) {  Loader::Types::User.new(id: 'alice') }
+
+  describe '.check_user_creation_allowed' do
+    context "Check user creation verification" do
+
+      it "should not allow user creation not under root" do
```

Comment:
Even better, if we use a subject we can remove a lot of this with something like:
```ruby
  describe '.check_user_creation_allowed' do
    context 'when the user is loaded in the "my_sub_tree" policy' do
      subject { Loader::Types::User.new(id: 'alice@my_sub_tree').check_user_creation_allowed }

      it { is_expected.not_to raise_error(Exceptions::InvalidPolicyObject)}
    end
  
    context 'when the user is loaded into the "root" policy' do
      subject { Loader::Types::User.new(id: 'alice') }
      allow(ENV).to receive(:[]).with('CONJUR_ALLOW_USER_CREATION').and_return(setting)

      context "when the CONJUR_ALLOW_USER_CREATION environment variable is false" do
        let(:setting) { 'false' }
        
        it { is_expected.to raise_error(Exceptions::InvalidPolicyObject)}
      end

      context "when the CONJUR_ALLOW_USER_CREATION environment variable is true" do
        let(:setting) { 'true' }
        
        it { is_expected.to raise_error(Exceptions::InvalidPolicyObject)}
      end
      ...
    end
  end

```

Reply:
Getting error on above:
 `allow` is not available on an example group (e.g. a `describe` or `context` block). It is only available from within individual examples (e.g. `it` blocks) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc).